### DISCUSSION TITLE
How to set particles' positions relative to the object in a behavior1k task?

### DISCUSSION THREAD

**Discussion by shengyin1224** [StanfordVL/BEHAVIOR-1K#1035] [Q&A] [answered]
I find the json file for a task in a certain scene like Beechwood_0_int_task_clean_a_leather_belt_0_0_template.json.

I want to know how can I set the value for keys in "system_registry", like 'water', 'cloth' or 'dust'. For example, there may be 20 dust particles. How to give an accurate value of positions? 

I find four obj states sampling functions -- fill, cover, soak, boil. Are these particles set by these sampling functions?

**Comment by hang-yin** [comment]
Hi @shengyin1224 , during sampling, I don't think we have precise control over where specifically the particles can go. 
However, after you load your scene, you can try to do something like 
```
system_name = "water"
water = scene.system_registry("name", system_name)
current_particle_positions = water.particle_positions
water.particle_positions = ${new particle positions}
```
@ChengshuLi can probably also comment more on this.

**Comment by ChengshuLi** [comment, marked as answer]
@shengyin1224 To add to Hang's comments, `Filled` and `Covered` states do spawn particles in relationship to other rigid objects.

You can also call the low level APIs to directly spawn particles. The following code will spawn two water particles at the specified positions.

```
 water_system = env.scene.get_system("water")
 water_system.generate_particles(positions=torch.tensor([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]]))
```

**Reply by shengyin1224** [reply]
thanks so much
